backends/gdx-backend-headless/src/com/badlogic/gdx/backends/headless/ZipResourceFile.java

```diff
@@ -0,0 +1,389 @@
+		 * doing an initial minimal read; if we don't find it, retry with a
+		 * second read as above.)
+		 */
+
+		// EOCD == 0x50, 0x4b, 0x05, 0x06
+		int eocdIdx;
+		for (eocdIdx = buffer.length - kEOCDLen; eocdIdx >= 0; eocdIdx--) {
+			if (buffer[eocdIdx] == 0x50
+					&& bbuf.getInt(eocdIdx) == kEOCDSignature) {
+				if (LOGV) {
+				}
+				break;
+			}
+		}
+
+		if (eocdIdx < 0) {
+		}
+
+		/*
+		 * Grab the CD offset and size, and the number of entries in the
+		 * archive. After that, we can release our EOCD hunt buffer.
+		 */
+
+		int numEntries = bbuf.getShort(eocdIdx + kEOCDNumEntries);
+		long dirSize = bbuf.getInt(eocdIdx + kEOCDSize) & 0xffffffffL;
+		long dirOffset = bbuf.getInt(eocdIdx + kEOCDFileOffset) & 0xffffffffL;
+
+		// Verify that they look reasonable.
+		if (dirOffset + dirSize > fileLength) {
+			throw new IOException();
+		}
+		if (numEntries == 0) {
+		}
+
+		if (LOGV) {
+
+		}
+
+		MappedByteBuffer directoryMap = f.getChannel().map(
+				FileChannel.MapMode.READ_ONLY, dirOffset, dirSize);
+		directoryMap.order(ByteOrder.LITTLE_ENDIAN);
+
+		byte[] tempBuf = new byte[0xffff];
+
+		/*
+		 * Walk through the central directory, adding entries to the hash table.
+		 */
+
+		int currentOffset = 0;
+
+		/*
+		 * Allocate the local directory information
+		 */
+		ByteBuffer buf = ByteBuffer.allocate(kLFHLen);
+		buf.order(ByteOrder.LITTLE_ENDIAN);
+
+		for (int i = 0; i < numEntries; i++) {
+			if (directoryMap.getInt(currentOffset) != kCDESignature) {
+				throw new IOException();
+			}
+
+			/* useful stuff from the directory entry */
+			int fileNameLen = directoryMap
+					.getShort(currentOffset + kCDENameLen) & 0xffff;
+			int extraLen = directoryMap.getShort(currentOffset + kCDEExtraLen) & 0xffff;
+			int commentLen = directoryMap.getShort(currentOffset
+					+ kCDECommentLen) & 0xffff;
+
+			/* get the CDE filename */
+
+			directoryMap.position(currentOffset + kCDELen);
+			directoryMap.get(tempBuf, 0, fileNameLen);
+			directoryMap.position(0);
+
+			/* UTF-8 on Android */
+			String str = new String(tempBuf, 0, fileNameLen);
+			if (LOGV) {
+			}
+
+			ZipEntryRO ze = new ZipEntryRO(zipFileName, file, str);
+			ze.mMethod = directoryMap.getShort(currentOffset + kCDEMethod) & 0xffff;
+			ze.mWhenModified = directoryMap.getInt(currentOffset + kCDEModWhen) & 0xffffffffL;
+			ze.mCRC32 = directoryMap.getLong(currentOffset + kCDECRC) & 0xffffffffL;
+			ze.mCompressedLength = directoryMap.getLong(currentOffset
+					+ kCDECompLen) & 0xffffffffL;
+			ze.mUncompressedLength = directoryMap.getLong(currentOffset
+					+ kCDEUncompLen) & 0xffffffffL;
+			ze.mLocalHdrOffset = directoryMap.getInt(currentOffset
+					+ kCDELocalOffset) & 0xffffffffL;
+
+			// set the offsets
+			buf.clear();
+			ze.setOffsetFromFile(f, buf);
+
+			// put file into hash
+			mHashMap.put(str, ze);
+
+			// go to next directory entry
+			currentOffset += kCDELen + fileNameLen + extraLen + commentLen;
+		}
+		if (LOGV) {
```

Comment:
Empty block: is it on purpose?

Reply:
Yeah, but I ought to remove them since they're just empty space.